Add examples for Artemis connector and fix issues in tests (#14769)
ballerina-platform/ballerina-lang: stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/Module.md

## Module overview

> This module provides the capability to connect with a ActiveMQ Artemis broker using the [Core API](https://activemq.apache.org/artemis/docs/latest/core.html)

Reviewer: ```suggestion
This module provides the capability to connect with a ActiveMQ Artemis broker using the [Core API](https://activemq.apache.org/artemis/docs/latest/core.html).
```

## Samples

### Artemis Producer

> Following program will produce a message to the Artemis broker

Reviewer: ```suggestion
The following program produces a message to the Artemis broker.
```

```ballerina
import ballerina/artemis;
import ballerina/log;

artemis:Producer prod = new({host:"localhost", port:61616}, "demo");
public function main() {
    var err = prod->send("Hello World!");
    if (err is error) {
        log:printError("Error occurred while sending message", err = err);
    }
    // Close the producer
    if (!prod.isClosed()) {
        err = prod->close();
        if (err is error) {
            log:printError("Error occurred closing producer", err = err);
        }
    }
}
```
### Artemis consumer

> Following program will consume a message from an Artemis broker

Reviewer: ```suggestion
The following program will consume a message from the Artemis broker.
```
